aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -315,5 +315,44 @@ Tensor& matmul_out(Tensor &result, const Tensor & tensor1, const Tensor & tensor
   return result;
 }
 
+Tensor matrix_power(const Tensor& a, int64_t n) {
+  AT_CHECK(a.dim() >= 2 && at::isFloatingType(a.type().scalarType()),
+           "matrix_power(", a.type(), "{", a.sizes(), "}): expected a tensor "
+           "of floating types with dim at least 2");
+  if (n == 0) {
+    return a.clone().copy_(at::eye(a.size(-2), a.options()).expand_as(a));
```

Comment:
`a.clone().copy_` is two copies, when you can just get by with 1 copy. What you want is `at::empty_like(a)` maybe?

Reply:
Oh, yes. Thank you. I will fix it right away.

Reply:
Based on the test failure, I think doing this will detach `a` from the autograd graph. Is there any other way to keep `a` linked to the graph?